wcivf/apps/elections/helpers.py

```diff
@@ -12,10 +13,40 @@ class EEHelper:
 
     ee_cache = {}
 
+    @property
+    def base_elections_url(self):
+        """
+        Builds the URL for the elections endpoint of EveryElection API
+        """
+        return f"{settings.EE_BASE}/api/elections/"
+
+    @cached_property
+    def deleted_election_ids(self):
+        """
+        Returns a generator object that retreives ids for elections that have
+        been deleted
+        """
+        url = f"{self.base_elections_url}?deleted=1"
+        pages = JsonPaginator(page1=url, stdout=sys.stdout)
+        for page in pages:
+            for result in page["results"]:
+                yield result["election_id"]
+
+    def delete_deleted_elections(self):
+        """
+        Deletes elections that were soft-deleted in EveryElection
+        """
+        from elections.models import Election  # noqa avoid circular import
+
+        return Election.objects.filter(
```

Comment:
`self.deleted_election_ids` will also include ballots (`group_type=None`), and we do sometimes delete a single ballot. In fact, this might be the most common path: for example when creating the IDs, we normally get the list of organisations right, but might make a mistake when adding by selecting "ward A" rather than "ward B" (this normally expresses itself as "Foo North ward" rather than "Foo North West Ward").

Relying on the CASCADE to delete the elections is smart when we've created the wrong _organisation_ group level election (`local.foo.2021-05-06`), but this doesn't address the ballot level soft deletes (`local.foo.foo-north.2021-05-06`).

I think adding the same query but for `PostElection` would deal with this.

Reply:
I've updated to attempt to delete PostElection objects also. But in the process I was trying to figure out how/where `group_type` is set, and noticed there is an `identifier_type` [field](https://github.com/DemocracyClub/EveryElection/blob/5145c0bd6138555b65cd1c8c0f1f4193f9890b9a/every_election/apps/elections/models.py#L300) that is exposed over the API. So I wonder if rather than use whole list of `election_id` values that come back, would it be better to filter them by their identifier type before making the filter/delete?